### PR TITLE
RockTheVote: add support for extended report

### PR DESCRIPTION
**Pull request by dannyboy15**
This adds the option to request the extended report which add a few extra fields. In addition, it adds support for the testing url.

h/t @natashamathur

### CODE REVIEW

**Comment by natashamathur** [comment]
For context, the reason I set up the report_type parameter the way it is was because

The existing code did not include a report_type parameter, and I wanted to make sure requests without that parameter still went through to minimize downstream effects and
I wasn't sure if RTV was going to release more report types, or if it was always going to be just "extended" vs "normal/default".

Overall, this should add the ability to use the testing environment, and to pull an "extended" report.

File: parsons/rockthevote/rtv.py [hunk 1]
```diff
@@ -41,26 +44,36 @@ class RockTheVote:
...
         self.partner_api_key = check_env.check('RTV_PARTNER_API_KEY', partner_api_key)
 
-        self.client = APIConnector('https://register.rockthevote.com/api/v4/',
-                                   headers=REQUEST_HEADERS)
+        if testing:
+            self.client = APIConnector(
+                "https://staging.rocky.rockthevote.com/api/v4", headers=REQUEST_HEADERS)
+        else:
+            self.client = APIConnector(
+                "https://register.rockthevote.com/api/v4", headers=REQUEST_HEADERS)
```
Comment (jburchard) [inline parsons/rockthevote/rtv.py:62]: Super picky, but can you move these URIs to the top of the script?

```
TESTING_URI = "https://staging.rocky.rockthevote.com/api/v4"
PRODUCTION_URI = "https://register.rockthevote.com/api/v4"
```

File: parsons/rockthevote/rtv.py [hunk 2]
```diff
@@ -41,26 +44,36 @@ class RockTheVote:
...
             before: str
-                Date before which to return registrations for
+                Limit to registrations that were started before this date, in
+                ISO format (e.g. 2020-01-01)
             since: str
-                Date for filtering registrations
+                Limit to registrations that were started since this date, in
+                ISO format (e.g. 2020-01-01)
+            report_type: str
+                The type of report to create. Currently only accepts "extended".
```
Comment (jburchard) [inline parsons/rockthevote/rtv.py:76]:
```
Currently only accepts ``extended``.
```
Also, could you explain what happens in the docs when it is left as ``None``?

File: parsons/rockthevote/rtv.py [hunk 3]
```diff
@@ -196,18 +223,27 @@ def run_registration_report(self, before=None, since=None, poll_interval_seconds
...
             before: str
-                Date before which to return registrations for
+                Limit to registrations that were started before this date, in
+                ISO format (e.g. 2020-01-01)
             since: str
-                Date for filtering registrations
+                Limit to registrations that were started since this date, in
+                ISO format (e.g. 2020-01-01)
+            report_type: str
+                The type of report to run. Currently only accepts "extended".
```
Comment (jburchard) [inline parsons/rockthevote/rtv.py:232]: Currently only accepts ``extended``.
```
Also, could you explain what happens in the docs when it is left as ``None``?

**Review by jburchard — CHANGES_REQUESTED** [review]
This is really great. Just a few minor comments.

**Review by jburchard — APPROVED** [review]

**Pull request merged**